src/app/inbound/mod.rs

```diff
@@ -146,33 +156,30 @@ pub fn spawn<P>(
     let dst_router = dst_stack
         .push_buffer_pending(max_in_flight, DispatchDeadline::extract)
         .push(router::layer(
-            router::Config::new("in dst", capacity, max_idle_age),
+            router::Config::new(capacity, max_idle_age),
             |req: &http::Request<_>| {
-                let canonical = req
-                    .headers()
+                req.headers()
                     .get(super::CANONICAL_DST_HEADER)
-                    .and_then(|dst| dst.to_str().ok())
-                    .and_then(|d| Addr::from_str(d).ok());
-                debug!("inbound canonical={:?}", canonical);
-
-                let dst = canonical
+                    .and_then(|dst| {
+                        dst.to_str().ok().and_then(|d| {
+                            Addr::from_str(d).ok().map(|a| {
+                                debug!("using {}", super::CANONICAL_DST_HEADER);
+                                a
+                            })
+                        })
+                    })
                     .or_else(|| {
                         super::http_request_l5d_override_dst_addr(req)
+                            .ok()
                             .map(|override_addr| {
-                                debug!("inbound dst={:?}; dst-override", override_addr);
+                                debug!("using l5d-dst-override");
```

Comment:
Nit/TIOLI: we use the value of `CANONICAL_DST_HEADER` const in the above message, should we use the dst override const here as well?

Reply:
Nit: should we still record the value of `override_addr` as well?
```suggestion
                                debug!(?override_addr, "using l5d-dst-override");
```